lib/crypto.js

```diff
@@ -562,9 +557,10 @@ exports.pbkdf2 = function(password,
 
 
 exports.pbkdf2Sync = function(password, salt, iterations, keylen, digest) {
-  if (typeof digest === 'undefined') {
-    digest = undefined;
-    pbkdf2DeprecationWarning();
+  if (digest === undefined) {
+    // Formerly Deprecation code DEP0009
+    throw new TypeError(
+      'The "digest" argument is required and must not be undefined');
```

Comment:
and `null` is allowed? Or will another layer throw for that case? basically, looks like the check should be against `== undefined` to me, but there is clearly another layer doing a check for digests like `'the-happy-digest'`, and other invalid input (numbers, etc.), so maybe its covered there.